lib/improver/utilities/temporal_interpolation.py

```diff
@@ -0,0 +1,443 @@
+
+            time_entry = initial_time
+            while True:
+                time_entry = (time_entry +
+                              timedelta(minutes=self.interval_in_minutes))
+                if time_entry >= final_time:
+                    break
+                time_list.append(time_entry)
+
+        return [('time', time_list)]
+
+    @staticmethod
+    def check_cube_coords_dtype(cube, new_cube, dim_coords=None):
+        """
+        Update the data type of the coords within the new_cube
+        to those in the original cube. This can be limited to a list of
+        dimension coordinates and associated auxillary coordinates if required.
+
+        Args:
+        cube (iris.cube.Cube):
+            The original cube that will be used to checked against for the
+            required data type of that coordinate in the new_cube.
+        new_cube (iris.cube.Cube):
+            The cube that must be checked and adjusted using the coordinates
+            from the original cube.
+        dim_coords (list):
+            List of dimension coordinate names. This list will be used to
+            limit the correction of the coordinate data type
+            to these coordinates and associated auxillary coordinates.
+            Default is None.
+
+        Returns:
+        new_cube (iris.cube.Cube):
+            Modified cube with the relevant coordinates datatypes
+            set to the same as on the original cube.
+
+        Raises:
+            CoordinateNotFoundError : If original cube does not have the
+                coordinate in dim_coords.
+            CoordinateNotFoundError : If coordinate in original cube is
+                not in new_cube.
+        """
+        if dim_coords is None:
+            coord_list = [coord.name() for coord in cube.coords()]
+        else:
+            coord_list = []
+            for coord_name in dim_coords:
+                try:
+                    req_dim = cube.coord_dims(coord_name)
+                    associated_coords = [crd.name() for crd in cube.coords()
+                                         if cube.coord_dims(crd) == req_dim
+                                         and crd.name() is not coord_name]
+                    coord_list.append(coord_name)
+                    for val in associated_coords:
+                        coord_list.append(val)
+                except CoordinateNotFoundError:
+                    msg = ('Original cube does not have the coordinate in'
+                           ' dim_coords {}'.format(coord_name))
+                    raise CoordinateNotFoundError(msg)
+        for coord_name in coord_list:
+            dtype_orig = cube.coord(coord_name).points.dtype
+            try:
+                new_coord = new_cube.coord(coord_name)
+                # TODO: Provide safe support for time coordinates that could
+                # potentially be in units of "hours since 1970-01-01 00:00:00."
+                if new_coord.points.dtype != dtype_orig:
```

Comment:
Is this the correct condition (here and at line 212)?  What if the new coordinate is also in float hours?  Don't we want to cast to int64 and seconds anyway?  Why are we looping over all coordinates here ... why are we not just looking at time / frt?

Reply:
@cgsandford Would you prefer if the `check_cube_coords_dtype` function became something like this:
```python
def check_time_coords_dtype(cube):
    for coord_name in ["time", "forecast_reference_time"]:
        coord = cube.coord(coord_name)
        coord.convert_units(
            "seconds since 1970-01-01 00:00:00")
        coord.points = np.around(new_coord.points)
        coord.points = coord.points.astype(np.int64)
        if (hasattr(coord, "bounds")
                and coord.bounds is not None)):
            coord.convert_units(
                "seconds since 1970-01-01 00:00:00")
            coord.bounds = np.around(coord.bounds)
            coord.bounds = coord.bounds.astype(np.int64)
    return cube
```

Reply:
Yes I would.  That wasn't why I asked this particular question, but it would solve the general problem.

Reply:
I've now modified this function.